.github/workflows/ui-version-number.yml

```diff
@@ -0,0 +1,60 @@
+name: FOLIO UI Calculate CI Version Number
+
+on:
+  workflow_call:
+    outputs:
+      version-number:
+        description: If a tag, we'll return that tag's version number; otherwise, it will be the derived version number unique to this run
+        value: ${{ jobs.get-version-number.outputs.version-number }}
+
+jobs:
+  get-version-number:
+    name: Get CI version number
+    runs-on: ubuntu-latest
+
+    outputs:
+      version-number: ${{ steps.final.outputs.version }}
+
+    steps:
+      - name: Checkout ${{ github.repository }}
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+          show-progress: false # spammy
+
+      - id: package-version
+        name: Get declared version from package.json
+        run: echo version=`jq -r .version package.json` | tee -a $GITHUB_OUTPUT
+
+      - name: Checkout folio-org/folio-tools
+        if: ${{ !startsWith(github.ref, 'refs/tags/v') }}
```

Comment:
This expression get used _a lot_. Two questions:
* should it be a shared function instead, like `isReleaseTag()`? 
* Do you think "starts with `v`" is sufficient? Can/should we do something more thorough like "matches `/^v[0-9]\.[0-9]\.[0-9]/` so we don't run this for, e.g., `very-interesting-tag` or `valery-test`?

Reply:
Unfortunately, there's not a great way to really do that.  The available functions are [extremely limited](https://docs.github.com/en/actions/learn-github-actions/expressions#functions), with no way to do any kind of regex/pattern matching.

I've moved all this into a job which runs a small script to calculate `is-release` which can then be passed around, as well as deriving the NPM registry to use